client/grpc/grpc_test.go

```diff
@@ -97,24 +104,81 @@ func TestDialContext(t *testing.T) {
 
 func TestSayHello(t *testing.T) {
 	mtr := mocktracer.New()
-	defer mtr.Reset()
 	opentracing.SetGlobalTracer(mtr)
+
 	ctx := context.Background()
-	conn, err := DialContext(ctx, "bufnet", grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
+	conn, err := DialContext(ctx, target, grpc.WithContextDialer(bufDialer), grpc.WithInsecure())
 	require.NoError(t, err)
 	defer func() {
 		require.NoError(t, conn.Close())
 	}()
 
 	client := examples.NewGreeterClient(conn)
-	resp, err := client.SayHello(ctx, &examples.HelloRequest{Firstname: "John", Lastname: "Doe"})
-	require.NoError(t, err)
-	assert.Equal(t, "Hello John Doe", resp.GetMessage())
-	expected := map[string]interface{}{
-		"component": "grpc-client",
-		"error":     false,
-		"span.kind": ext.SpanKindEnum("producer"),
-		"version":   "dev",
+
+	tt := []struct {
```

Comment:
I would suggest to use a map based table test. https://dave.cheney.net/2019/05/07/prefer-table-driven-tests